pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -281,6 +282,14 @@ in
         singularity = finalAttrs.finalPackage;
       };
     };
+    # Run the packages to test. E.g.
+    # nix run .#apptainer.manualTests.run-image-hello-cowsay
+    manualTests = {
+      run-image-hello-cowsay = writeShellScriptBin "test-run-image-hello-cowsay" ''
+        ${lib.getExe finalAttrs.finalPackage} exec ${finalAttrs.passthru.tests.image-hello-cowsay} hello \
+          | ${lib.getExe finalAttrs.finalPackage} exec ${finalAttrs.passthru.tests.image-hello-cowsay} cowsay
```

Comment:
Maybe

```
run-image-hello-cowsay =
  let
    singularity = getExe finalAttrs.finalPackage;
    image = finalAttrs.tests.image-hello-cowsay
  in ...
```

Also it's still not clear to me what is it we expect from passthru tests in the cross-compilation configuration, but let's limit our focus to native builds for now...

Reply:
Addressed.